core/policy.go

```diff
@@ -81,14 +77,16 @@ func (p *DefaultQueuePolicy) HandleNewHead(ctx context.Context, target PolicyTar
 	}
 
 	// Expire messages that have been in the queue for too long; they will probably never be mined.
-	height, err := newHead.Height()
-	if err != nil {
-		return err
-	}
-	if height >= p.maxAgeRounds { // avoid uint subtraction overflow
-		expired := target.ExpireBefore(ctx, (height - p.maxAgeRounds))
-		for _, msg := range expired {
-			log.Errorf("Outbound message %v expired un-mined after %d rounds", msg, p.maxAgeRounds)
+	if len(newTips) > 0 {
+		height, err := newTips[0].Height()
+		if err != nil {
+			return err
+		}
+		if height >= p.maxAgeRounds { // avoid uint subtraction overflow
+			expired := target.ExpireBefore(ctx, (height - p.maxAgeRounds))
+			for _, msg := range expired {
+				log.Errorf("Outbound message %v expired un-mined after %d rounds", msg, p.maxAgeRounds)
```

Comment:
(I know you haven't changed this here, but..) Does this warrant an error message -- could this be a warning instead?